src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt

```diff
@@ -0,0 +1,38 @@
+package com.hibob.academy.filter
+
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.ext.Provider
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+const val CREATE_TOKEN_PATH = "createToken"
+const val COOKIE_NAME = "JWT"
+
+@Component
+@Provider
+class AuthenticationFilter: ContainerRequestFilter {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    override fun filter(requestContext: ContainerRequestContext) {
+
+        if (requestContext.uriInfo.path != CREATE_TOKEN_PATH) {
+            verify(requestContext.cookies[COOKIE_NAME]?.value)
+            logger.info("Authorized successfully")
+        }
+    }
+
+    fun verify(cookie: String?): Jws<Claims> {
```

Comment:
@elijulian - @IsraelCohen54  changed the code as you requested
I'm approving and if you have more coomets please add comments and a new PR will be done with changes
I want to let @IsraelCohen54  continue  with his following PRs